src/nominatim_api/results.py

```diff
@@ -717,6 +722,28 @@ async def complete_linked_places(conn: SearchConnection, result: BaseResult) ->
         result.linked_rows.append(_result_row_to_address_row(row))
 
 
+async def complete_entrances_details(conn: SearchConnection, results: List[BaseResultT]) -> None:
+    """ Retrieve information about tagged entrances for this place.
+    """
+    t = conn.t.place_entrance
+    sql = sa.select(t.c.place_id, t.c.osm_id, t.c.type, t.c.location, t.c.extratags)\
+            .where(t.c.place_id.in_(r.place_id for r in results))
```

Comment:
You can filter results here for `result.source_table == SourceTable.PLACEX`. Results from other tables won't have entrances.